backend/src/api/model/series.rs

```diff
@@ -11,71 +11,124 @@ use crate::{
             event::{Event, EventSortOrder}
         },
         Node,
-        NodeValue,
     },
     db::types::{SeriesState as State, Key},
     prelude::*,
 };
 
 
-pub(crate) struct Series {
+#[graphql_interface(Context = Context, for = [WaitingSeries, ReadySeries])]
+pub(crate) trait Series {
+    // TODO Shared like for blocks??
+    fn id(&self) -> Id;
+    fn opencast_id(&self) -> &str;
+
+    /// Returns a list of realms where this series is referenced (via some kind of block).
+    async fn host_realms(&self, context: &Context) -> ApiResult<Vec<Realm>> {
+        let cols = Realm::col_names("realms");
+        let query = format!("\
+            select {cols} \
+            from realms \
+            inner join blocks \
+                on realms.id = blocks.realm_id and
+                type = 'series' and blocks.series_id = $1 \
+        ");
+        context.db.query_mapped(
+            &query,
+            dbargs![&self.id().key_for(Id::SERIES_KIND)],
+            Realm::from_row
+        )
+            .await?
+            .pipe(Ok)
+    }
+}
+
+pub(crate) struct SharedData {
     pub(crate) key: Key,
     opencast_id: String,
-    state: State,
-    title: Option<String>,
-    description: Option<String>,
 }
 
-impl Node for Series {
+pub(crate) struct WaitingSeries {
+    pub(crate) shared: SharedData,
+}
+
+impl Series for WaitingSeries {
     fn id(&self) -> Id {
-        Id::series(self.key)
+        Id::series(self.shared.key)
+    }
+
+    fn opencast_id(&self) -> &str {
+        &self.shared.opencast_id
     }
 }
 
-#[graphql_object(Context = Context, impl = NodeValue)]
-impl Series {
+#[graphql_object(Context = Context, impl = SeriesValue)]
+impl WaitingSeries {
     fn id(&self) -> Id {
-        Node::id(self)
+        Series::id(self)
     }
 
-    fn state(&self) -> State {
-        self.state
+    fn opencast_id(&self) -> &str {
+        self.opencast_id()
```

Comment:
It's not because the declarations in these `graphql_*` `impl`-s don't really end up a s `fn`-s, right? They are arms in a big `match` or whatever. But now that you mention it, yeah, explicitly referring to the trait is probably a good idea here. :thinking:

Reply:
Ye, we also do `Series::id(self)` above. But ye.